### PR TITLE
Prevent checkstyle violations through Travis

### PR DESCRIPTION
**Pull request by Blaisorblade**
Now that I fixed all violations, enable checkstyle on Travis.

* Turn Checkstyle violations into errors.
* Run Checkstyle on Travis.
* Update the uploaded style for Android Studio.

### CODE REVIEW

**Comment by Blaisorblade** [comment]
Since this would affect all future PRs, I'm interested in opinions of all current contributors (I'm not proposing this requires a unanimity).
@nahojjjen @sarav @DJCrapsody @oskomonzon

**Comment by sarav** [comment]
My only nitpick is generally line size. But we agreed to disagree there. So, as long as this enforces style as we edit and doesn't break android studio, I'm all for it.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -35,6 +42,9 @@ Set-up Git on Android Studio to automate Git commands. For developers who are *l
 
 ## Contributing with Pull Requests
 
+### Ensure Commits Follow Coding Style
+Use Android Studio autoformatting and check with CheckStyle that your PR does not add coding style violations.
```
Comment (Blaisorblade) [inline DEVELOPMENT.md]: Should this be more tentative? "Try to ensure that..."?

**Comment by Blaisorblade** [comment]
I've added instructions for configuring Android Studio. Note I had to change the coding style to make this work.

Also note: reformat will *not* fix all errors—the style file has ideas on names and Javadocs.
I still think this workflow is worth a try.

**Pull request merged**